recipes/llvm/all/conanfile.py

```diff
@@ -0,0 +1,791 @@
+            'with_ffi': [True, False],
+            'with_zlib': [True, False],
+            'with_xml2': [True, False],
+            'with_curl': [True, False],
+            'with_zstd': [True, False],
+            'with_httplib': [True, False],
+            'keep_binaries_regex': ['ANY'],
+
+            # options removed in package id
+            'use_llvm_cmake_files': [True, False],
+            'ram_per_compile_job': ['ANY'],
+            'ram_per_link_job': ['ANY'],
+            # conan center index ci workaround, memory consumption to high in debug builds
+            # and only latest versions are enabled in ci to reduce ci runtime by days
+            'conan_center_index_limits': [True, False],
+            'enable_unsafe_mode': [True, False],
+        },
+    }
+    default_options = {
+        **{
+            'with_project_' + project: project in default_projects
+            for project in projects
+        },
+        **{
+            "with_runtime_" + runtime: runtime in default_runtimes
+            for runtime in runtimes
+        },
+        **{
+            'shared': False,
+            'llvm_build_llvm_dylib': False,
+            'llvm_link_llvm_dylib': False,
+            'llvm_dylib_components': 'all',
+            'fPIC': True,
+            'targets': 'all',
+            'exceptions': True,  # llvm 14 default off
+            'rtti': True,  # llvm 14 default off
+            'threads': True,
+            'lto': 'Off',
+            'static_stdlib': False,
+            'unwind_tables': True,
+            'expensive_checks': False,
+            'use_perf': False,
+            'llvm_use_sanitizer': '',
+            'with_z3': False,
+            'with_ffi': False,
+            'with_zlib': True,
+            # TODO curl 7.76.0 check is failing in config-ix.cmake
+            'with_curl': False,
+            'with_zstd': True,
+            'with_httplib': False,
+            # TODO default True, issues with liblldb.so and conan: Findlibxml2.cmake not provided
+            'with_xml2': False,
+            'keep_binaries_regex': '^$',
+
+            # options removed in package id
+            # XXX Should these files used by conan at all?
+            'use_llvm_cmake_files': False,
+            # creating job pools with current free memory
+            'ram_per_compile_job': '2000',
+            'ram_per_link_job': '14000',
+            'conan_center_index_limits': True,
+            'enable_unsafe_mode': False,
+        }
+    }
+
+    exports_sources = 'patches/**/*', 'conandata.yml'
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        apply_conandata_patches(self)
+
+    def is_windows(self):
+        return self.settings.os == "Windows"
+
+    def is_macos(self):
+        return self.settings.os == "MacOS"
+
+    def is_linux(self):
+        return self.settings.os == "Linux"
```

Comment:
To be safe, you should prefix any custom methods or attributes with an `_` to avoid conflicts.